src/backend/routers/agent.py

```diff
@@ -81,45 +82,34 @@ async def create_agent(
     user_id = ctx.get_user_id()
     logger = ctx.get_logger()
 
-    agent_data = AgentModel(
-        name=agent.name,
-        description=agent.description,
-        preamble=agent.preamble,
-        temperature=agent.temperature,
-        user_id=user_id,
-        organization_id=agent.organization_id,
-        tools=agent.tools,
-        is_private=agent.is_private,
-    )
     deployment_db, model_db = get_deployment_model_from_agent(agent, session)
     try:
-        created_agent = agent_crud.create_agent(session, agent_data)
-
-        if agent.tools_metadata:
-            for tool_metadata in agent.tools_metadata:
-                await update_or_create_tool_metadata(
-                    created_agent, tool_metadata, session, ctx
-                )
-
         if deployment_db and model_db:
-            deployment_config = (
-                agent.deployment_config
-                if agent.deployment_config
-                else deployment_db.default_deployment_config
-            )
-            agent_crud.assign_model_deployment_to_agent(
-                session,
-                agent=created_agent,
-                deployment_id=deployment_db.id,
-                model_id=model_db.id,
-                deployment_config=deployment_config,
-                set_default=True,
+            agent_data = AgentModel(
+                name=agent.name,
+                description=agent.description,
+                preamble=agent.preamble,
+                temperature=agent.temperature,
+                user_id=user_id,
+                organization_id=agent.organization_id,
+                tools=agent.tools,
+                is_private=agent.is_private,
+                deployment_id=deployment_db.id if deployment_db else None,
```

Comment:
Instead of defaulting to None, can we fallback to Cohere Platform + command-r-plus as defaults?

Reply:
Thank you, I've added defaults for deployment and model.